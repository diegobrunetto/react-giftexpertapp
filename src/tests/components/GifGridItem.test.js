import React from "react";
const { shallow } = require("enzyme");
const { GifGridItem } = require("../../components/GifGridItem");

describe("Pruebas en GifGridItem", () => {
  const title = "titulo";
  const url = "https://localhost/algo.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe tener un texto con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe tener la imagen igual al url y alt", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
});
