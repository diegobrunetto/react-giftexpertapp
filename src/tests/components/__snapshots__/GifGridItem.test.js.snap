// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn nes-container is-rounded "
>
  <img
    alt="titulo"
    className="image"
    src="https://localhost/algo.jpg"
  />
  <p
    className="title"
  >
    titulo
  </p>
</div>
`;
